src/communication/response_broker.hpp

```diff
@@ -194,17 +194,22 @@ namespace cubcomm
   }
 
   template <typename T_PAYLOAD, typename T_ERROR>
-  void
+  T_ERROR
   response_broker<T_PAYLOAD, T_ERROR>::bucket::register_request (response_sequence_number a_rsn)
   {
     {
       std::lock_guard<std::mutex> lk_guard (m_mutex);
 
-      assert (!m_terminate);
+      if (m_terminate)
+	{
+	  return m_error;
```

Comment:
Does it return ERROR_ON_WRITE as initial value?

Reply:
Yes, it is. It's the same as what get_response() below returns if it spot the termination.